File: src/main/java/com/example/service/AdminUserService.java
package com.example.service;

import com.example.dto.request.UserFilterDTO;
import com.example.dto.response.AdminUserResponseDTO;
import com.example.dto.response.AdminUserStatsDTO;
import org.springframework.data.domain.Page;

import java.util.List;

public interface AdminUserService {

    // ================= USERS =================

    Page<AdminUserResponseDTO> getAllUsers(
            UserFilterDTO filter,
            int page,
            int size,
            String sortBy,
            String direction
    );

    AdminUserResponseDTO getUserById(Long id);

    List<AdminUserResponseDTO> getActiveUsers();

    List<AdminUserResponseDTO> searchUsers(String keyword);

    // ================= USER MANAGEMENT =================

    AdminUserResponseDTO activateUser(Long id);

    AdminUserResponseDTO deactivateUser(Long id);

    AdminUserResponseDTO verifyEmail(Long id);

    AdminUserResponseDTO verifyPhone(Long id);

    AdminUserResponseDTO blockUser(Long id);

    AdminUserResponseDTO unblockUser(Long id);

    void softDeleteUser(Long id);

    void deleteUser(Long id);

    // ================= DASHBOARD =================

    AdminUserStatsDTO getUserStatistics();
    void bulkActivateUsers(List<Long> userIds);
    void bulkBlockUsers(List<Long> userIds);
    void bulkUnblockUsers(List<Long> userIds);
    void bulkSoftDeleteUsers(List<Long> userIds);
    AdminUserResponseDTO restoreUser(Long id);
    byte[] exportUsersToCsv();
    byte[] exportUsersToExcel();
}

File: admin panel/my-react-app/src/services/adminUserService.js
import { apiClient } from "./api";

/* ==========================================================
   USER LIST
========================================================== */

export const getUsers = async (
  page = 0,
  size = 10,
  search = "",
  filters = {}
) => {

  let endpoint = `/admin/users?page=${page}&size=${size}`;

  if (search.trim()) {
    endpoint += `&search=${encodeURIComponent(search)}`;
  }

  Object.entries(filters).forEach(([key, value]) => {
    if (value !== undefined && value !== null && value !== "") {
      endpoint += `&${key}=${encodeURIComponent(value)}`;
    }
  });

  return await apiClient(endpoint);
};

/* ==========================================================
   USER DETAILS
========================================================== */

export const getUserById = async (id) => {
  return await apiClient(`/admin/users/${id}`);
};

/* ==========================================================
   USER ACTIONS
========================================================== */

export const activateUser = async (id) => {
  return await apiClient(`/admin/users/${id}/activate`, {
    method: "PUT",
  });
};

export const deactivateUser = async (id) => {
  return await apiClient(`/admin/users/${id}/deactivate`, {
    method: "PUT",
  });
};

export const blockUser = async (id) => {
  return await apiClient(`/admin/users/${id}/block`, {
    method: "PUT",
  });
};

export const unblockUser = async (id) => {
  return await apiClient(`/admin/users/${id}/unblock`, {
    method: "PUT",
  });
};

export const verifyEmail = async (id) => {
  return await apiClient(`/admin/users/${id}/verify-email`, {
    method: "PUT",
  });
};

export const verifyPhone = async (id) => {
  return await apiClient(`/admin/users/${id}/verify-phone`, {
    method: "PUT",
  });
};

export const restoreUser = async (id) => {
  return await apiClient(`/admin/users/${id}/restore`, {
    method: "PUT",
  });
};

export const softDeleteUser = async (id) => {
  return await apiClient(`/admin/users/${id}/soft-delete`, {
    method: "PUT",
  });
};

export const deleteUser = async (id) => {
  return await apiClient(`/admin/users/${id}`, {
    method: "DELETE",
  });
};

/* ==========================================================
   BULK ACTIONS
========================================================== */

export const bulkActivateUsers = async (userIds) => {
  return await apiClient("/admin/users/bulk/activate", {
    method: "PUT",
    body: JSON.stringify({ userIds }),
  });
};

export const bulkBlockUsers = async (userIds) => {
  return await apiClient("/admin/users/bulk/block", {
    method: "PUT",
    body: JSON.stringify({ userIds }),
  });
};

export const bulkUnblockUsers = async (userIds) => {
  return await apiClient("/admin/users/bulk/unblock", {
    method: "PUT",
    body: JSON.stringify({ userIds }),
  });
};

export const bulkSoftDeleteUsers = async (userIds) => {
  return await apiClient("/admin/users/bulk/soft-delete", {
    method: "PUT",
    body: JSON.stringify({ userIds }),
  });
};

/* ==========================================================
   EXPORT
========================================================== */

export const exportUsersCsv = async () => {
  return await apiClient("/admin/users/export/csv", {
    method: "GET",
  });
};

export const exportUsersExcel = async () => {
  return await apiClient("/admin/users/export/excel", {
    method: "GET",
  });
};